src/AWS.Deploy.Common/Exceptions.cs

```diff
@@ -318,5 +318,26 @@ public static string PrettyPrint(this Exception? e)
 
             return $"{Environment.NewLine}{e.Message}{Environment.NewLine}{e.StackTrace}{PrettyPrint(e.InnerException)}";
         }
+
+        /// <summary>
+        /// Returns the truncated error message by only preserving the leading and trailing k characters.
+        /// The message will only be truncated if its length is greater than 2*k.
+        /// This method returns <see cref="string.Empty"/> if the error message is null or empty.
+        /// </summary>
+        public static string GetTruncatedErrorMessage(this Exception? e, int k = 500)
```

Comment:
Please document parameters using the XML tags, and there's usually a better parameter name than a single letter.

Reply:
Done